prompterator/models/openai_models.py

```diff
@@ -89,42 +88,102 @@ def __init__(self):
 
         super().__init__()
 
+    @staticmethod
+    def get_function_calling_tooling_name(json_schema):
+        function = json_schema.copy()
+        return function.pop("title")
+
+    @staticmethod
+    def build_function_calling_tooling(json_schema, function_name):
+        """
+        @param function_name: name for the openai tool
+        @param json_schema: contains desired output schema in proper Json Schema format
+        @return: list[tools] is (a single function in this case) callable by OpenAI model
+            in function calling mode.
+        """
+        function = json_schema.copy()
+        description = (
+            function.pop("description")
+            if function.get("description", None) is not None
+            else function_name
+        )
+        tools = [
+            {
+                "type": "function",
+                "function": {
+                    "name": function_name,
+                    "description": description,
+                    "parameters": function,
+                },
+            }
+        ]
+
+        return tools
+
+    @staticmethod
+    def build_response_format(json_schema):
+        """
+        @param json_schema: contains desired output schema in proper Json Schema format
+        @return: dict with desired response format directly usable with OpenAI API
+        """
+        schema = {"name": json_schema.pop("title"), "schema": json_schema, "strict": True}
```

Comment:
Just to be sure: You use `pop` because you want to remove the entry, right? Otherwise, you could use `get`...

Reply:
yes